app/uk/gov/hmrc/gform/views/summary/TextFormatter.scala

```diff
@@ -18,40 +18,71 @@ package uk.gov.hmrc.gform.views.summary
     textConstraint match {
-      case PositiveNumber(_, fracDigits, rm, unit) => formatNumber(currentValue, fracDigits, rm, unit)
-      case Number(_, fracDigits, rm, unit)         => formatNumber(currentValue, fracDigits, rm, unit)
-      case s: Sterling                             => formatSterling(currentValue)
-      case _                                       => currentValue
+      // format: off
+      case IsPositiveNumberOrNumber(PositiveNumberOrNumber(_, maxFractionalDigits, roundingMode, unit)) => formatNumber(currentValue, maxFractionalDigits, roundingMode, unit)
+      case _: Sterling                                                                                  => formatSterling(currentValue)
+      case _                                                                                            => currentValue
+      // format: on
     }
 
   def componentTextEditable(currentValue: String, textConstraint: TextConstraint): String =
     textConstraint match {
-      case PositiveNumber(_, _, _, unit) => stripTrailingZeros(currentValue)
-      case Number(_, _, _, unit)         => stripTrailingZeros(currentValue)
-      case s: Sterling                   => stripTrailingZeros(currentValue)
-      case _                             => currentValue
+      // format: off
+      case IsPositiveNumberOrNumber(PositiveNumberOrNumber(_, _, _, _)) => stripTrailingZeros(currentValue)
+      case _: Sterling                                                  => stripTrailingZeros(currentValue)
+      case _                                                            => currentValue
+      // format: on
+    }
+
+  def componentTextForSummary(
+    currentValue: String,
+    textConstraint: TextConstraint,
+    prefix: Option[SmartString],
+    suffix: Option[SmartString])(
+    implicit l: LangADT
+  ): String =
+    (textConstraint, prefix, suffix) match {
+      // format: off
+      case (IsPositiveNumberOrNumber(PositiveNumberOrNumber(_, maxFractionalDigits, roundingMode, unit)), p, s)  =>   prependPrefix(p) + formatNumber(currentValue, maxFractionalDigits, roundingMode,  s.map(_.localised).orElse(unit))
+      case (_: Sterling, _, _)                                                                                   =>   formatSterling(currentValue)
+      case (_, p, s)                                                                                             =>   prependPrefix(p) + currentValue + appendSuffix(s)
+      case _                                                                                                     =>   currentValue
+      // format: on
     }
 
   private def stripTrailingZeros(currentValue: String): String =
     if (currentValue.contains(".")) {
       currentValue.reverse.dropWhile(_ == '0').dropWhile(_ == '.').reverse
     } else currentValue
 
+  private def prependPrefix(
+    prefix: Option[SmartString]
+  )(
+    implicit l: LangADT
+  ): String =
+    prefix.fold("")(_.localised.value + " ")
```

Comment:
There is `value` method available for `SmartString`, so instead of `_.localised.value` use `_.value`

Reply:
Changed.